packages/office-ui-fabric-react/src/components/ScreenReaderAlert/ScreenReaderAlert.tsx

```diff
@@ -0,0 +1,117 @@
+/**
+ * @copyright Microsoft Corporation. All rights reserved.
+ *
+ * @file Screen Reader Alert component.
+ */
+
+import * as React from 'react';
+
+import { IScreenReaderAlertProps, defaultScreenReaderAlertProps, ReadingMode } from './ScreenReaderAlert.Props';
+import './ScreenReaderAlert.scss';
+
+/**
+ * This is a screen reader alert component for developers to easily add screen reader feature to there web site.
+ * This component is a wrapper, you would like to wrap the texts for screen reader to read.
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions Live Region in MDN}
+ *
+ * All functionalities of this component is tested through:
+ * 1. JAWS 18.
+ * 2. NVDA 2016.4.
+ * 3. ChromeVOX v53. (Not support ReadingMode.ReadAfterOtherContent option for ChromeVOX, it will read immediately)
+ * 4. Narrator in windows 10.
+ *
+ * @example
+ * <ScreenReaderAlert>
+ *   { yourFirstAlertText }
+ *   { yourSecondAlertText }
+ * </ScreenReaderAlert>
+ */
+export class ScreenReaderAlert extends React.Component<IScreenReaderAlertProps, {}> {
+  public static defaultProps: IScreenReaderAlertProps = defaultScreenReaderAlertProps;
+
+  /**
+   * The index updated during each rendering to make React re-attach the DOM.
+   *
+   * It is critical to make React re-attach the DOM once we want it to read something. Simply updating the
+   * text context inside DOM will cause different behavior in different screen readers or browsers.
+   * Re-attaching the live region element is the most common way which is supported by most major screen reader
+   * and browser combinations.
+   */
+  private _renderIndex: number = 0;
+
+  constructor(props: IScreenReaderAlertProps) {
+    super(props);
+  }
+
+  public shouldComponentUpdate(nextProps: IScreenReaderAlertProps): boolean {
+    return (this.props.indicator !== nextProps.indicator || !this._isSameText(this.props, nextProps));
+  }
+
+  public render(): JSX.Element {
+    return (
+      <div className='ms-ScreenReaderAlert' key={ this._renderIndex++ }>
+        {
+          this.props.readingMode !== ReadingMode.DoNotRead && (
+            <p
+              role={ this._role }
+              aria-live={ this._ariaLive }
+              aria-atomic={ true }>
+              { this._alertMessages }
```

Comment:
Notifications are ephemeral in nature. They shouldn't persist indefinitely. As it is, this message will be readable in DOM and can won't get removed. Imagine the scenario where you're updating fields and you add an alert for saved or success. You can modify form more, and the message will still be in DOM and an AT user reading through will hit the text again., but it's no longer relevant or accurate, leading to confusion. 

We've done 3 for toast notifications in the UI before. This is how sharepoints toast notifications ("saving...", "checking out...", etc) work.